HeterogeneousCore/SonicTriton/src/TritonClient.cc

```diff
@@ -306,10 +385,10 @@ void TritonClient::evaluate() {
       return;
   } else {
     //blocking call
-    tc::InferResult* results;
+    std::vector<tc::InferResult*> results;
     success = handle_exception([&]() {
       TRITON_THROW_IF_ERROR(
-          client_->Infer(&results, options_, inputsTriton_, outputsTriton_, headers_, compressionAlgo_),
+          client_->InferMulti(&results, options_, inputsTriton, outputsTriton, headers_, compressionAlgo_),
           "evaluate(): unable to run and/or get result");

```

Comment:
Based on the code on `getResults()` I guess `results` is the owner of the pointers it contains. How about converting the vector of raw pointers to vector of smart pointers already here to make make their ownership clear.

Reply:
The interface that requires `std::vector<InferResult*>* results` is set by Triton: https://github.com/triton-inference-server/client/blob/r22.08/src/c%2B%2B/library/grpc_client.cc#L1132

Reply:
I understand the `InferMulti()` needs the vector of raw pointers, but how about converting it to vector of smart pointers right after the call returns?